### PR TITLE
Import projects from Data Management Plan files

### PR DESCRIPTION
**Pull request by Menjabin**
Adds a workflow for importing projects from Data Management Plan files that follow the RDA DMP Common Standard. This format is used by e.g. ELIXIR's Data Stewardship Wizard, and it would greatly simplify the process of publishing projects from there to a SEEK instance if it could be done by a simple file upload rather than by manually entering the data.

This solution reads the data about a project from a DMP file, and this is used to create a new project. All contributors to the project are also read from the file, and user accounts are created and added as members of the newly created project.
The option to import a project is currently only presented to the user on both the "new project" and "guided create" pages. It is not accessible through the main navbar.

### CODE REVIEW

**Comment by stuzart** [comment]
Hi,
There is something wrong with the PR and cannot review or merge. It contains many changes to many files, many of them reverting recent changes. I've tried looking at the commit itself by cloning and checking outside of github - and it is the same there too.
It's possible that resyncing with the latest main branch may solve things, but otherwise it will require a new PR that contains only the changes for the new feature.

**Comment by Menjabin** [comment]
My apologies,
I have updated the branch so that only my changes remain.

**Review by stuzart — CHANGES_REQUESTED** [review]
it lacks any functional or unit tests, and due to being quite complex and involved it really needs.
Without them it is difficult to review the pull request works, and also there is high potential for things to break without knowing with future changes

**Comment by Menjabin** [comment]
I have now added tests for the new functionality

**Review by stuzart — APPROVED** [review]

File: config/routes.rb [hunk 1]
```diff
@@ -737,7 +742,7 @@
       post :template_attributes
     end
     collection do
-      post :filter_isa_tags_by_level
+post :filter_isa_tags_by_level
```
Comment (kdp-cloud) [inline config/routes.rb:745]: Missing indentation?

**Review by kdp-cloud — APPROVED** [review]
There's a missing indentation in `routes.rb` on line 745.
Other than that, no comments!

File: config/routes.rb (hunk 1 above)
Comment (stuzart) [reply config/routes.rb:745]: we can fix that after it's merged

**Comment by stuzart** [comment]
thanks for the contribution!

**Pull request merged**